udf/udf_safe_geo_struct.sql

```diff
@@ -0,0 +1,26 @@
+CREATE TEMP FUNCTION
+  udf_safe_geo_struct(country STRING,
+    city STRING,
+    geo_subdivision1 STRING,
+    geo_subdivision2 STRING) AS ( --
+    IF(country IS NULL
+      OR country = '??',
+      NULL,
+      STRUCT(country,
+        NULLIF(city,
+          '??') AS city,
+        NULLIF(geo_subdivision1,
+          '??') AS geo_subdivision1,
+        NULLIF(geo_subdivision2,
+          '??') AS geo_subdivision2)));
+
+/*
+
+Convert geoip lookup fields to a struct, replacing '??' with NULL.
+
+Returns NULL if if required field country would be NULL.
```

Comment:
@jklukas r? on this function rename (from `udf_require_whole_geo`)

Reply:
The `safe_` prefix _almost_ fits, since this will return null on invalid input, but it implies there's an error we're handling, which isn't really true here. I was trying to figure out what might be "dangerous" in a naive implementation of this function because of the name.

Some other options:

udf_mk_geo
udf_geo_of
udf_geo_from
udf_parse_geo
udf_build_geo